### DISCUSSION TITLE
MultiZarrToZarr: Concatenate along dimensionless variable/coordinate

### DISCUSSION THREAD

**Discussion by mpvginde** [fsspec/kerchunk#502] [Q&A] [answered]
Hi,

I'm trying to combine netCDF-files with 3 dimensions `[x, y, leadtime]` along a new dimension 'reftime'.
The `reftime` of each file is included in the netCDF as a dimensionless coordinate as seen here: 
![image](https://github.com/user-attachments/assets/443c2ebc-3f42-4ed4-a4ed-ce443ea383d6)

Typically I open the multiple files with xarray like this:
```python
def preprocess(x):
    x = x.expand_dims(dim="reftime")
    return x
ds = xr.open_mfdataset(files,preprocess=preprocess)
```

What would be the best way to use the reftime value of each file as value of the `concat_dim`?
Do I add the new dimension already when creating single file reference, or do I use some preprocessing or coo_map when using `MultiZarrToZar`?

Kind regards,
Michiel

**Comment by martindurant** [comment]
> Do I add the new dimension already when creating single file reference, or do I use some preprocessing or coo_map

I would say this is a matter of choice - does the individual dataset gain anything by adding in the extra coordinate information? If no (my suspicion), then using coo_map or a preprocess function in multizarrtozarr is totally fine, probably better the former.

**Reply by mpvginde** [reply]
Thanks @martindurant, I agree.
Any tips on how you could achieve this? 
In other words how can I access the actual value of reftime using the coo_map? 
The tutorials show examples for accessing filenames, but it is unclear to me how I can retrieve and use the data of a coordinate of a file as values for the new dimension.

**Reply by martindurant** [reply, marked as answer]
The mapper variant you want is "data:{var}" (where var is "reftime"), or you can always fall back to the completely general function `F(index, fs, var, fn)`, where you would need to open the zarr dataset in the function and extract the value.

**Reply by mpvginde** [reply]
Thanks a lot!
